src/Interpreters/ProcessList.cpp

```diff
@@ -564,7 +564,7 @@ bool QueryStatus::checkTimeLimitSoft()
     if (is_killed.load())
         return false;
 
-    return limits.checkTimeLimit(watch.elapsedNanoseconds(), OverflowMode::BREAK);
+    return needs_to_stop_by_timeout;
```

Comment:
For some reason, I cannot do `return needs_to_stop_by_timeout;` because clickhouse-test doesn't start:
```
Traceback (most recent call last):
  File "/home/ubuntu/ClickHouse/./tests/clickhouse-test", line 3714, in <module>
    main(args)
  File "/home/ubuntu/ClickHouse/./tests/clickhouse-test", line 2962, in main
    args.build_flags = collect_build_flags(args)
  File "/home/ubuntu/ClickHouse/./tests/clickhouse-test", line 2541, in collect_build_flags
    value = int(
ValueError: invalid literal for int() with base 10: b''
```

Only this line causes this error. The reason is not clear to me.